Rivet/Functions/Import-RivetPlugin.ps1

```diff
@@ -5,32 +5,54 @@ function Import-RivetPlugin
     param(
         [Parameter(Mandatory)]
         [AllowEmptyCollection()]
-        [String[]]$Path
+        [String[]]$Path,
+
+        [Parameter(Mandatory)]
+        [AllowEmptyCollection()]
+        [String[]]$ModuleName
     )
 
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
     
-    # $DebugPreference = 'Continue'
-
     Write-Timing -Message 'Import-RivetPlugin  BEGIN' -Indent
 
-    $commands = & {
+    $moduleNames = & {
         foreach( $pluginPath in $Path )
         {
             if( [IO.Path]::GetExtension($pluginPath) -eq '.ps1' )
             {
-                Write-Error -Message ('Unable to import Rivet plugin "{0}": invalid plugin file extension. A Rivet plugin must be a PowerShell module. The path to your plugin must be to a directory that is importable by the `Import-Module` command, or to a .psd1 or .psm1 file.' -f $pluginPath) -ErrorAction $ErrorActionPreference
+                Write-Error -Message ('Unable to import Rivet plugin "{0}": invalid plugin file extension. A Rivet plugin must be a PowerShell module. The path to your plugin must be to a directory that is importable by the `Import-Module` command, or to a .psd1 or .psm1 file.' -f $pluginPath) -ErrorAction Stop
                 continue
             }
 
-            Write-Timing -Message ('  Import {0}  BEGIN' -f $pluginPath)
-            $module = Import-Module -Name $pluginPath -Global -Force -PassThru
-            Write-Timing -Message ('  Import {0}  END' -f $pluginPath)
+            Write-Timing -Message "  Import  BEGIN  $($pluginPath)"
+            Import-Module -Name $pluginPath -Global -Force -PassThru |
+                Select-Object -ExpandProperty 'Name' |
+                Write-Output
+            Write-Timing -Message "  Import  END    $($pluginPath)"
+
+        }
+
+        $ModuleName | Write-Output

```

Comment:
Debugging or expected all time?

Reply:
It's just making it explicit that the module name is being returned from the current block to the outer block.

```
$moduleNames = & {
    # snip
    $ModuleName | Write-Output
}
```